docs/nix-develop.md

```diff
@@ -0,0 +1,206 @@
+# Using Nix Develop Environment
```

Comment:
How users would access this content? we can add a new entry on `mkdocs.yaml` or reference it from an existing docs file or both